### ISSUE TITLE
Preferences Dialog crashes Geany

### ISSUE THREAD

**Issue by jrittenh** [codebrainz/overview-plugin#7]
Trying to open the preferences dialog crashes geany with the error below:

```
ERROR:overviewui.c:216:overview_ui_supports_left_position: assertion failed: (DOC_VALID (doc))
```

Running Gentoo, Geany-1.24.1

**Comment by codebrainz** [comment]
Do you by any chance have that option enable that allows there to be no documents open? (as opposed to what I think is the default where it always keeps an "untitled" document open after the last document is closed).

**Comment by jrittenh** [comment]
It appears so (I can close the last document and Geany stays open but doesn't open a new "Untitled" document), although I can't find the setting for that option anywhere anymore.  I actually don't even see it in geany.conf, although that doesn't mean I didn't miss it.

**Comment by jrittenh** [comment]
A sanitized geany.conf if it helps:

```
[geany]
default_open_path=
cmdline_new_files=true
notebook_double_click_hides_widgets=false
tab_close_switch_to_mru=false
tab_pos_sidebar=2
sidebar_pos=0
highlighting_invert_all=false
pref_main_search_use_current_word=true
check_detect_indent=false
detect_indent_width=false
use_tab_to_indent=true
pref_editor_tab_width=4
indent_mode=2
indent_type=0
virtualspace=1
autocomplete_doc_words=false
completion_drops_rest_of_word=false
autocompletion_max_entries=30
autocompletion_update_freq=250
color_scheme=
mru_length=10
disk_check_timeout=30
show_editor_scrollbars=true
brace_match_ltgt=false
use_gtk_word_boundaries=true
complete_snippets_whilst_editing=false
use_atomic_file_saving=false
gio_unsafe_save_backup=false
use_gio_unsafe_file_saving=false
indent_hard_tab_width=8
find_selection_type=0
extract_filetype_regex=-\\*-\\s*([^\\s]+)\\s*-\\*-
show_symbol_list_expanders=true
compiler_tab_autoscroll=true
allow_always_save=false
statusbar_template=
new_document_after_close=false
msgwin_status_visible=true
msgwin_compiler_visible=true
msgwin_messages_visible=true
msgwin_scribble_visible=true
documents_show_paths=true
sidebar_page=0
pref_main_load_session=false
pref_main_project_session=true
pref_main_project_file_in_basedir=false
pref_main_save_winpos=true
pref_main_confirm_exit=false
pref_main_suppress_status_messages=false
switch_msgwin_pages=false
beep_on_errors=true
auto_focus=false
sidebar_symbol_visible=true
sidebar_openfiles_visible=true
editor_font=Monospace 10
tagbar_font=Sans 9
msgwin_font=Sans 9
show_notebook_tabs=true
show_tab_cross=true
tab_order_ltr=true
tab_order_beside=false
tab_pos_editor=2
tab_pos_msgwin=0
use_native_windows_dialogs=false
show_indent_guide=true
show_white_space=true
show_line_endings=false
show_markers_margin=true
show_linenumber_margin=true
long_line_enabled=true
long_line_type=0
long_line_column=80
long_line_color=#C2EBC2
symbolcompletion_max_height=10
symbolcompletion_min_chars=3
use_folding=true
unfold_all_children=false
use_indicators=true
line_wrapping=false
auto_close_xml_tags=true
complete_snippets=true
auto_complete_symbols=true
pref_editor_disable_dnd=false
pref_editor_smart_home_key=true
pref_editor_newline_strip=false
line_break_column=80
auto_continue_multiline=true
comment_toggle_mark=~ 
scroll_stop_at_last_line=true
autoclose_chars=0
pref_editor_default_new_encoding=UTF-8
pref_editor_default_open_encoding=None
default_eol_character=2
pref_editor_new_line=true
pref_editor_ensure_convert_line_endings=true
pref_editor_replace_tabs=false
pref_editor_trail_space=false
pref_toolbar_show=true
pref_toolbar_append_to_menu=false
pref_toolbar_use_gtk_default_style=true
pref_toolbar_use_gtk_default_icon=true
pref_toolbar_icon_style=0
pref_toolbar_icon_size=0
pref_template_developer=
pref_template_company=
pref_template_mail=
pref_template_initial=
pref_template_version=1.0
pref_template_year=%Y
pref_template_date=%Y-%m-%d
pref_template_datetime=%d.%m.%Y %H:%M:%S %Z
context_action_cmd=
sidebar_visible=true
statusbar_visible=true
msgwindow_visible=true
fullscreen=false
scribble_text=Type here what you want, use it as a notice/scratch board
scribble_pos=57
treeview_position=273
msgwindow_position=686
geometry=0;0;1920;1030;1;
custom_date_format=
custom_commands=sort;uniq;tr [:lower:] [:upper:];tr [:upper:] [:lower:];
custom_commands_labels=sort;uniq;upper;lower;
msgwin_orientation=1

[build-menu]
number_ft_menu_items=0
number_non_ft_menu_items=0
number_exec_menu_items=0

[search]
pref_search_hide_find_dialog=false
pref_search_always_wrap=false
pref_search_current_file_dir=true
find_all_expanded=true
replace_all_expanded=true
position_find_x=1069
position_find_y=326
position_replace_x=706
position_replace_y=428
position_fif_x=648
position_fif_y=399
fif_regexp=false
fif_case_sensitive=true
fif_match_whole_word=false
fif_invert_results=false
fif_recursive=false
fif_extra_options=
fif_use_extra_options=false
fif_files=
fif_files_mode=0
find_regexp=true
find_case_sensitive=false
find_escape_sequences=true
find_match_whole_word=false
find_match_word_start=false
find_close_dialog=false
replace_regexp=false
replace_case_sensitive=true
replace_escape_sequences=true
replace_match_whole_word=false
replace_match_word_start=false
replace_search_backwards=false
replace_close_dialog=false

[plugins]
load_plugins=true
custom_plugin_path=
active_plugins=/usr/lib64/geany/addons.so;/usr/lib64/geany/debugger.so;/usr/lib64/geany/geanyvc.so;/usr/lib64/geany/htmlchars.so;/usr/lib64/geany/pretty-printer.so;

[tools]
term_cmd=xterm
browser_cmd=firefox
grep_cmd=grep
terminal_cmd=xterm -e "/bin/sh %c"

[printing]
print_cmd=/usr/bin/lpr %f
use_gtk_printing=true
print_line_numbers=true
print_page_numbers=true
print_page_header=true
page_header_basename=false
page_header_datefmt=%c

[VTE]
load_vte=true
emulation=xterm
send_selection_unsafe=false
send_cmd_prefix=
font=Monospace 10
scroll_on_key=true
scroll_on_out=true
enable_bash_keys=true
ignore_menu_bar_accel=false
follow_path=false
run_in_vte=false
skip_run_script=false
cursor_blinks=false
scrollback_lines=500
shell=/bin/bash
colour_fore=#FFFFFF
colour_back=#000000
last_dir=
image=

[project]
session_file=
project_file_path=

[files]
recent_files=
recent_projects=
current_page=-1
```

**Comment by codebrainz** [comment]
It's the "new_document_after_close=false" line that controls that preference. It's stashed away under the "Various" tab in the Preferences dialog. Anyway, I have removed the debug assertion in b194e8c0595c333bf8a1c84aa9c038ce7116fb96 so it doesn't abort anymore, see the commit message for caveat.

**Comment by codebrainz** [comment]
BTW, I might have changed the preference from the default, I'm not sure and don't feel like rummaging through Geany's source to see :)

**Comment by codebrainz** [comment]
Closing since it's fixed in b194e8c0595c333bf8a1c84aa9c038ce7116fb96.
